docs/user-guide/aws-examples/eks.rst

```diff
@@ -0,0 +1,215 @@
+
+.. code-block:: yaml
+
+    cat <<EOF | kubectl apply -f -
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRole
+    metadata:
+      name: pod-exec
+    rules:
+    - apiGroups:
+      - ""
+      resources:
+      - pods
+      - pods/exec
+      verbs:
+      - list
+      - get
+      - watch
+      - create
+    ---
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRoleBinding
+    metadata:
+      name: allow-pods-exec
+    subjects:
+    - kind: ServiceAccount
+      name: default
+      namespace: default
+    roleRef:
+      kind: ClusterRole
+      name: pod-exec
+      apiGroup: rbac.authorization.k8s.io
+    EOF
+
+The above yaml file creates a ClusterRole and a ClusterRoleBinding.
+
+ClusterRole Definition:
+
+- Specifies permissions (rules) for interacting with Kubernetes pods.
+- resources: ["pods", "pods/exec"] specifies the resources pods and pods/exec.
+- verbs: ["list", "get", "watch", "create"] lists the actions allowed on these resources (list, get, watch, create).
+
+ClusterRoleBinding Definition:
+
+- Binds the pod-exec ClusterRole to a specific ServiceAccount (default in the default namespace).
+- This means that any pods using the default ServiceAccount in the default namespace will have the permissions specified in the pod-exec ClusterRole.
+
+ Now, we can spin up a client pod.
+
+.. code-block:: yaml
+
+    cat <<EOF | kubectl apply -f -
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: client-pod
+      labels:
+        app: client
+    spec:
+      containers:
+        - name: client
+          image: python:3.10-slim-bullseye
+          command: ["sh", "-c", "pip install kubernetes && sleep infinity"]
+    EOF
+
+Here, we are using a light-weight public python docker image and installing kubernetes Python client package so that we can run kubeclient.py from this client pod and connect to the scheduler pod to run existing NeMo modules.
+
+Once the client-pod is up and running we can copy the kubeclient.py script into the client pod and and run the the script.
+
+.. code-block:: bash
+
+    kubectl cp examples/k8s/kubeclient.py client-pod:kubeclient.py
+    kubectl exec client-pod -- python3 kubeclient.py --command "add_id --scheduler-address localhost:8786 --input-data-dir=/nemo-workspace/arxiv --output-data-dir=/nemo-workspace/arxiv-addid/"
+
+
+This approach is allows the execution of NeMo modules within the scheduler pod from a separate client pod. This separation ensures that the client pod can be provisioned with specific permissions tailored for executing commands and accessing resources within the Kubernetes environment.
+
+Moreover, deploying this client pod can be orchestrated by another service such as AWS Batch, facilitating scalable and efficient management of computational tasks within Kubernetes clusters.
```

Comment:
Thank you for this explanation! This makes a lot of sense. Do you mind moving it to before you spin up the client pod so the user has background into why they would be doing this?